### DISCUSSION TITLE
How do I get tabbed toc selections to NOT change the tiddler shown in other toc rendered in the stream?

### DISCUSSION THREAD

**Discussion by Tejeev** [TiddlyWiki/TiddlyWiki5#5298] [Q&A]
If I've 3 tiddlers open, each with their own tabbed toc, whatever I select in one is displayed in the other two.  Is there something here that I can adjust to give each its own context?

**Comment by joshuafontany** [comment]
The `toc-tabbed-` macros use a default system tiddler to store the "currently selected tiddler".

https://tiddlywiki.com/#Table-of-Contents%20Macros
Under **Parameters**, we see
**selectedTiddler**
    The title of the state tiddler for noting the currently selected tiddler, defaulting to `$:/temp/toc/selectedTiddler`. It is recommended that this be a system tiddler.

As you are not overriding this parameter, the default is being used for all 3 toc-macro-instances. You need to give each of them a unique **selectedTiddler** value. One way of doing this is the `<<qualify>>` macro. 
https://tiddlywiki.com/#qualify%20Macro

In order to use a macro as a parameter value for another macro, you must call the macro via the `<$macrocall>` widget-syntax like so:
https://tiddlywiki.com/#MacroCallWidget
```
<$macrocall $name="toc-tabbed-external-nav" tag="MyToCTag" selectedTiddler=<<qualify "$:/state/toc/MyToCTag">> unselectedText="Please select from the Table of Contents. Click the arrow to expand a topic."/>
```

Each instance of this macrocall-widget in DIFFERENT TIDDLERS will have a unique numerical value appended to "$:/state/toc/MyToCTag" (the Qualify parameter), and thus each instance will keep track of the selected tiddler in unique locations and behave independently from each other.

Best,
Joshua Fontany
